### ISSUE TITLE
Implement drivers for the LAPIC and IOAPIC so that IRQs can be used

### ISSUE THREAD

**Issue by KonnerV** [CharlotteOS-Deprecated/charlotte-core#24]
Develop drivers to enable the configuration and use of the local and I/O APICs to send interrupt requests to the CPU.

**Comment by mdpcs3544** [comment]
Just wanted to clarify that this is the APIC not the legacy 8259 PIC.

**Comment by KonnerV** [comment]
Yea, APIC is what I thought you meant because the 8259 PIC is quite old

**Comment by mdpcs3544** [comment]
Perfect!

**Comment by EthanPlant** [comment]
Right now this is being blocked on #55

**Comment by mdpcs3544** [comment]
I unassigned this and set the status on the project board to blocked.

**Comment by mdpcs3544** [comment]
I've set the status to this to in progress since some parts have already been done and merged into main. I've assigned it to @xadaemon who did that work.

**Comment by xadaemon** [comment]
Ok, I'm back to working on charlotte this weekend, I want to maybe finish irq driver for good
